tuple.go

```diff
@@ -87,42 +88,38 @@ func (t Tuple) marshal(at uint16) (netfilter.Attribute, error) {
 }
 
 // An IPTuple encodes a source and destination address.
-// Both of its members are of type net.IP.
+// Both of its members are of type netip.Addr.
 type IPTuple struct {
-	SourceAddress      net.IP
-	DestinationAddress net.IP
+	SourceAddress      netip.Addr
+	DestinationAddress netip.Addr
 }
 
 // Filled returns true if the IPTuple's fields are non-zero.
 func (ipt IPTuple) filled() bool {
-	return len(ipt.SourceAddress) != 0 && len(ipt.DestinationAddress) != 0
+	return ipt.SourceAddress.IsValid() && ipt.DestinationAddress.IsValid()
 }
 
 // unmarshal unmarshals netlink attributes into an IPTuple.
-//
-// IPv4 addresses will be represented by a 4-byte net.IP, IPv6 addresses by 16-byte.
-// The net.IP object is created with the raw bytes, NOT with net.ParseIP().
-// Use IP.Equal() to compare addresses in implementations and tests.
 func (ipt *IPTuple) unmarshal(ad *netlink.AttributeDecoder) error {
 	if ad.Len() != 2 {
 		return errNeedChildren
 	}
 
 	for ad.Next() {
-		b := ad.Bytes()
-		if len(b) != 4 && len(b) != 16 {
+		addr, ok := netip.AddrFromSlice(ad.Bytes())
+		if !ok {
 			return errIncorrectSize
 		}
 
 		switch ipTupleType(ad.Type()) {
 		case ctaIPv4Src:
```

Comment:
Fold these cases like
```suggestion
		case ctaIPv4Src, ctaIPv6Src:
```